### PR TITLE
Setting DISCONT flag after seeking.

### PR DESCRIPTION
**Pull request by milos-pesic-sc**

### CODE REVIEW

**Comment by milos-pesic-sc** [comment]
@tchakabam @pokey909 
Setting discont flag on stream start causes decoder to drop that buffer so we have audio cut off. But not setting it (and/or unsetting it) after seek command causes unexpected behaviour in terms of state changes - so we end up with active playback session still going on but player is in BUFFERING state.
The solution here sets discont flag after seek commands and unsets it otherwise.

**Comment by pokey909** [comment]
so the initial assumption was actually right that we previously ran into problems when seeking without discont? I wonder if we are also dropping buffers after seeking without knowing it.
According to the docs, a discont might result in clearing some adapter buffers. 
What happens if we either set GST_BUFFER_FLAG_RESYNC alone or in combination with a discont?
From the docs it sounds like GST_BUFFER_FLAG_RESYNC tells the pipeline to resync the clocks at this timestamp.
If we also drop buffer after seeking, it should probably show as a short period of silence.

**Comment by milos-pesic-sc** [comment]
good point - will check raw audio signal after seek. GStreamer hlsdemux implementation uses GST_BUFFER_FLAG_DISCONT after seek command in a similar way as we do here. Doc is not super clear but for GST_BUFFER_FLAG_DISCONT it says "This typically occurs after a seek..".

**Comment by milos-pesic-sc** [comment]
@pokey909 
You were right - setting discont flag on seek causes dropping the first buffer coming after. When I use GST_BUFFER_FLAG_RESYNC instead of GST_BUFFER_FLAG_DISCONT - no dropping occurs.

File: src/skippy_hlsdemux.c [hunk 1]
```diff
@@ -916,10 +916,16 @@ skippy_hls_demux_proxy_pad_chain (GstPad *pad, GstObject *parent, GstBuffer *buf
   SkippyHLSDemux *demux = SKIPPY_HLS_DEMUX (gst_pad_get_element_private (pad));
 
   GST_OBJECT_LOCK (demux);
-  GST_BUFFER_FLAG_UNSET (buffer, GST_BUFFER_FLAG_DISCONT);
-  if (demux->need_segment) {
+  if (G_UNLIKELY(demux->need_stream_start)) {
```
Comment (tchakabam) [inline src/skippy_hlsdemux.c]: @milos-pesic-sc dude i wonder if this logic is more "compatable" but probably not :)

lgtm :+1:

**Pull request merged**